sopel/modules/url.py

```diff
@@ -436,32 +419,107 @@ def check_callbacks(bot: SopelWrapper, url: str, use_excludes: bool = True) -> b
     )
 
 
-def find_title(url: str, verify: bool = True) -> Optional[str]:
-    """Return the title for the given URL.
+def find_title(
+    url: str,
+    verify: bool = True,
+    allow_local: bool = False,
+    memory: tools.SopelMemory = {},
```

Comment:
I'd replace that by 2 parameters:

* `unsafe_urls`: a list of unsafe URLs
* `unsafe_domains`: a list of unsafe domain name

You would then need to call the function as follow:

```python
unsafe_urls = [
    unsafe_url
    for unsafe_url, safe_data in bot.memory.get("safety_cache", {}).items()
    if safe_data.get('positives', 0) > 0
]
unsafe_domains = bot.memory.get("safety_cache_local", {}).values()

result = find_title(url, verify, allow_local, unsafe_urls, unsafe_domains)
```

No need for a SopelMemory anymore, no need to pass the bot around, etc.

Reply:
Good point, thanks.